### DISCUSSION TITLE
Typescript help after Middleware configuration

### DISCUSSION THREAD

**Discussion by noobCode-69** [vercel/next.js#55510] [Help] [answered]
### Summary

Hey everyone , I am little new to Ts and Next middleware . 
I want some help , my company is currently migrating to typescipt

Question : 

I have defined a middleware where I made sure (100%) that router.locale is defined (router is basically returned from useRouter()) 
but the type of router.locale i get is "string | undefined" which is not expected since i made sure it's defined i want it to be "string"

Please Help!!
Thanks

**Comment by icyJoseph** [comment, marked as answer]
This is one of those situations where TypeScript can't really infer the type. Mostly because you have different processes at hand.

What you could do is, have a `useLocale` hook, which reads from the `router.locale`, and throws when that is `undefined`.

```js
// pseudo-code
export const useLocale = () => {
  const {locale} = router;
   if (typeof locale === 'undefined') {
      throw new Error("Locale was not set, double check middleware impl")
   }
   return locale;
}
```

And now TS can infer that `useLocale` never gives back undefined, and if locale ever fails, an error boundary or the dev-error overlay will catch it.
